### PR TITLE
Automatically upgrade new password hashes

### PR DESCRIPTION
**Pull request by Toflar**
This will automatically upgrade bcrypt passwords to argon2 passwords when users log in (or whatever we will have in the future). Symfony FTW :)

Implements https://github.com/contao/contao/issues/538.

### CODE REVIEW

**Review by bytehead — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

File: core-bundle/src/Security/User/ContaoUserProvider.php [hunk 1]
```diff
@@ -107,6 +107,19 @@ public function supportsClass($class): bool
         return $this->userClass === $class;
     }
 
+    /**
+     * @param User $user
+     */
+    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    {
+        if (!is_a($user, $this->userClass)) {
```
Comment (Toflar) [inline core-bundle/src/Security/User/ContaoUserProvider.php:115]: This shouldn't happen because of the `supports()` call. But anyway :)

Comment (leofeyer) [reply core-bundle/src/Security/User/ContaoUserProvider.php:115]: You are right. I have just copied the code from the `refreshUser()` method, but maybe it is unnecessary there as well?

https://github.com/contao/contao/blob/1e08698e6122029c6d8eda45010e59eb54bc2fcd/core-bundle/src/Security/User/ContaoUserProvider.php#L87-L91

Comment (Toflar) [reply core-bundle/src/Security/User/ContaoUserProvider.php:115]: Well, defensive programming - guess that's fine :)

**Pull request merged**